contracts/zero-ex/contracts/src/features/Migrate.sol

```diff
@@ -0,0 +1,94 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "../fixins/FixinOwnable.sol";
+import "../errors/LibOwnableRichErrors.sol";
+import "../storage/LibOwnableStorage.sol";
+import "../storage/LibMigrateStorage.sol";
+import "../migrations/LibMigrate.sol";
+import "../migrations/LibBootstrap.sol";
+import "./IFeature.sol";
+import "./IMigrate.sol";
+import "./ISimpleFunctionRegistry.sol";
+
+
+/// @dev Migration features.
+contract Migrate is
+    IFeature,
+    IMigrate,
+    FixinOwnable
+{
+    // solhint-disable const-name-snakecase
+
+    /// @dev Name of this feature.
+    string constant public override FEATURE_NAME = "Migrate";
+    /// @dev Version of this feature.
+    uint256 constant public override FEATURE_VERSION = (1 << 64) | (0 << 32) | (0);
+
+    /// @dev Initializes this feature.
+    /// @param impl The actual address of this feature contract.
+    /// @return success Magic bytes if successful.
+    function bootstrap(address impl) external returns (bytes4 success) {
+        // Register feature functions.
+        ISimpleFunctionRegistry(address(this)).extend(this.migrate.selector, impl);
+        ISimpleFunctionRegistry(address(this)).extend(this.getMigrationOwner.selector, impl);
+        return LibBootstrap.BOOTSTRAP_SUCCESS;
+    }
+
+    /// @dev Execute a migration function in the context of the ZeroEx contract.
+    ///      The result of the function being called should be the magic bytes
+    ///      0x2c64c5ef. Only callable by the owner.
```

Comment:
How are the magic bytes throughout these files generated? Are they fn selectors?

Reply:
It's the hash of `MIGRATE_SUCCESS` and `BOOTSTRAP_SUCCESS`. I never really understood the rationale behind using fn selectors. Wouldn't you want a distinct value that you can't accidentally return?

Reply:
Yeah, this works. Could we just add a cheeky comment about how it's generated so future devs know to use the same pattern?